src/main/java/org/jabref/gui/keyboard/KeyBindingRepository.java

```diff
@@ -113,8 +113,10 @@ public int size() {
 
     public Optional<KeyBinding> mapToKeyBinding(KeyEvent keyEvent) {
         for (KeyBinding binding : KeyBinding.values()) {
-            if (checkKeyCombinationEquality(binding, keyEvent)) {
-                return Optional.of(binding);
+            if (!binding.getDefaultKeyBinding().isEmpty()){
```

Comment:
Are you sure this works? I would gess `getDefaultKeyBinding` is always empty for `NEW_INBOOK` even though the user has overwritten it. On first glance, I would suggest to change the `getKeyCombination` method below to `public Optional<KeyCombination> getKeyCombination(KeyBinding bindName)` and return an empty optional if `binding` is empty.

Moreover, `get` should probably return en empty string instead of `"Not associated"` to be consistent.

Reply:
It works. If I change the next statement 
from `public KeyCombination getKeyCombination(KeyBinding bindName)` 
to `Optional<KeyCombination> getKeyCombination(KeyBinding bindName)` 

I have to change other parts of code. For example line 141 
from `KeyCombination keyCombination = getKeyCombination(binding);`
to `KeyCombination keyCombination = getKeyCombination(binding).get()`;

Is this what you mean?

Reply:
Okay, can you briefly explain why it works? I thought "getDefaultKeyBinding" then always returns an empty string, so this check is always false.

Yeah, there are more code changes required for my suggestion. I think it would provide a clean solution to handle the case that for a given keybinding there is not necessarily a keycombination. But I didn't had the time yet to check the code in detail. So if your solution indeed works, then it's fine with me as well.

Reply:
I started debugger and see that it always returns shortcut from `KeyBinding.java` file. 

<img width="1389" alt="Screenshot 2020-10-14 at 12 56 37" src="https://user-images.githubusercontent.com/20572030/95974019-0e331000-0e1d-11eb-9808-397b0022ca36.png">

Solution with `Optional` looks more clean. I just want to confirm that if we implement it, we also change other parts of code to `getKeyCombination(binding).get()`.

Reply:
In case this is done, the code fragment is `getKeyCombination(binding).ifpresent(combination -> ...)`

Reply:
@Gena928 Exactly! So if you change the keybinding for "New inbook" to something, then `getDefaultKeybinding` should still return an empty string, so the code never goes into the if body in this case (although it should to match the new-user defined key).

Reply:
@tobiasdiez 
I agree. If you change default shortcut, this code will not work.

Reply:
I'll try to fix today.